### ISSUE TITLE
Use with another app name or already initialised app

### ISSUE THREAD

**Issue by kjetilhp** [dappsnation/ngfire#15]
Question rather than issue I guess, is it possible with current version to provide an already initialised firebase app or give another name to the config? Trying to use it together with angularfire (and akita-ng-fire) while refactoring away from akita-ng-fire.

**Comment by GrandSchtroumpf** [comment]
I see. Unfortunately this would require a lot of changes. The goal of the lib was to not use `@angular/fire` at all.

Let's take your example : 
```typescript
app: (app: App) => getApp()
```
Where would the app in the param come from ? And how where would the firebase app be initialized for an app not using `@angular/fire` ?

-----

Just thinking here, but maybe I could let the developer specify the app s.he wants to initialize. If nothing is specified, we do as currently. It would look like that : 
```typescript
export const environment = {
  production: true,
  firebase: {
    options: {
      ...
    },
    /** 
    * If provided, will use the returned app instead of initializing the default app
    * @param options the same as provided above
    * @param name only if FIREBASE_APP_NAME provider exist in the injector
    */
    app: (options, name) => getApp('my-name'),
    storage: (app, storage) => getStorage(app, 'custom-storage')
  },
}
```

Then it means that the emulators will have to return the services (which would be a breaking change) :
```typescript
export const environment = {
  production: false,
  firebase: {
    options: {
      ...
    },
    firestore: (app, settings) => {
      const firestore = getFirebase(app, settings);
      useFirestoreEmulator(firestore);
      return firestore:
    },
    auth: (app, ) => {
      ...
    },
  },
}
```

Would it help ?

_Note_: 
It might create a lot of issues to mix the `@angular/fire` initializers in the config while using the firebase sdk (without the zone wrapper) in the `CollectionFire` service.

**Comment by kjetilhp** [comment]
I do not have the same knowledge around these libraries as you do so I'm not sure about issues that might arise, but regarding the config it looks to me that it would solve my issue, I might be naive though.  Regarding the getApp(), my understanding was that this returns the app from angular/fire (firebase) when already initialised.
```javascript
// angular/fire init..
provideFirebaseApp(() => {
      const app = initializeApp(environment.firebase)
      return app
    }),
    provideFirestore(() => {
      const firestore = getFirestore()
      connectFirestoreEmulatorInDevMode(firestore)
      return firestore
    }),
....
// after init the getApp() would return the instance needed for ng-fire.. I assume
```
If my request does not fit with your plans for the project, and causes unwanted side effects/changes we can close the issue.

**Comment by GrandSchtroumpf** [comment]
I'll give it a try and come back to you.

**Comment by GrandSchtroumpf** [comment]
@kjetilhp I've open a PR : https://github.com/dappsnation/ngfire/pull/16
Can you take a look and let me know if it solves this issue ?
You can try it out with : 
```
npm i ngfire@next
```

**Comment by kjetilhp** [comment]
@GrandSchtroumpf awesome, this works great! thanks

**Comment by GrandSchtroumpf** [comment]
@kjetilhp I'm glad it helped.
I just released version 0.0.39 on `latest`: `npm i ngfire@latest`
